Add git_credential_id attribute to databricks_repo resource (#5845)
databricks/terraform-provider-databricks: docs/resources/repo.md

* `url` -  (Required) The URL of the Git Repository to clone from. If the value changes, Git folder is re-created.
* `git_provider` - (Optional, if it's possible to detect Git provider by host name) case insensitive name of the Git provider.  Following values are supported right now (could be a subject for a change, consult [Repos API documentation](https://docs.databricks.com/dev-tools/api/latest/repos.html)): `gitHub`, `gitHubEnterprise`, `bitbucketCloud`, `bitbucketServer`, `azureDevOpsServices`, `gitLab`, `gitLabEnterpriseEdition`, `awsCodeCommit`.

> * `git_credential_id` - (Optional) The ID of the Git credential (`databricks_git_credential`) to use for this repo.  Required when the workspace has multiple credentials or when automatic credential selection fails (`GIT_CLI_CANNOT_CHOOSE_CREDENTIAL`).  If the value changes, the Git folder is re-created.

Reviewer: ```suggestion
* `git_credential_id` - (Optional) The ID of the Git credential (`databricks_git_credential`) to use for authentication with the Git provider remote.
```

let's just keep it plain and simple. we dont actually need to specify this param if you have multiple git credentials, you can still rely on git credential defaults (this was a bug before that we fixed, it was not working)
